test/validation/api/ipsec/ipsec.c

```diff
@@ -807,6 +807,13 @@ static int ipsec_send_out_one(const ipsec_test_part *part,
 	return num_out;
 }
 
+static void ipsec_pkt_proto_err_set(odp_packet_t pkt)
+{
+	uint8_t *data = (uint8_t *)odp_packet_data(pkt);
+	data += odp_packet_len(pkt) - 4;
+	data[0] = ~data[0];
```

Comment:
It would be good to have a comment what this does. This appears to me to write into the ICV area of the packet in most (all?) cases, so why do we not get an auth error instead of proto error?

This may access data outside the segment, depending on the packet length. Or maybe is won't if there is suitable pool config and packet construction elsewhere in the code. Using odp_packet_copy_from_mem() would be more robust.

Reply:
You are right (on both fronts). Will have these addressed in the next version.